pkg/admissioncontroller/webhooks/auth/seed/handler.go

```diff
@@ -50,15 +50,17 @@ func init() {
 }
 
 // NewHandler creates a new HTTP handler for authorizing requests for resources related to a Seed.
-func NewHandler(logger logr.Logger, authorizer auth.Authorizer) http.HandlerFunc {
+func NewHandler(ctx context.Context, logger logr.Logger, authorizer auth.Authorizer) http.HandlerFunc {
 	h := &handler{
+		ctx:        ctx,
 		logger:     logger,
 		authorizer: authorizer,
 	}
 	return h.Handle
 }
 
 type handler struct {
+	ctx        context.Context
```

Comment:
From https://pkg.go.dev/context
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx: 


If you really want to do this, please add a short explanation comment why it was done so.

Reply:
Reading through the reasons for the statement in https://go.dev/blog/context-and-structs, I think it would be somehow ok-ish in this case.
However, I found an even better way: we can just use the http request's context as a parent context for our timeout context. 
This is also what the controller-runtime webhook handlers are doing and I think it makes sense, so I adopted this.